akmod-specs/displaylink.spec

```diff
@@ -0,0 +1,138 @@
+%{_libexecdir}/%{name}/ella-dock-release.spkg
+%{_libexecdir}/%{name}/firefly-monitor-release.spkg
+%{_libexecdir}/%{name}/ridge-dock-release.spkg
+%dir /var/log/%{name}/
+%{_prefix}/lib/systemd/system-sleep/displaylink.sh
+
+%package config
+Summary:	Supporting config files for DisplayLink USB graphics adapters
+License:	MIT
+URL:		https://github.com/displaylink-rpm/displaylink-rpm
+BuildArch:	noarch
+Requires:	%{name}-manager logrotate
+
+%description config
+This adds support for HDMI/VGA adapters built upon the DisplayLink DL-6xxx,
+DL-5xxx, DL-41xx and DL-3xxx series of chip-sets. This includes numerous
+docking stations, USB monitors, and USB adapters.
+
+%preun config
+%systemd_preun displaylink.service
+
+%postun config
+%systemd_postun_with_restart displaylink.service
+
+%files config
+%config(noreplace) %{_sysconfdir}/logrotate.d/%{name}
+%{_unitdir}/displaylink.service
+%{_udevrulesdir}/99-displaylink.rules
+%{_datadir}/X11/xorg.conf.d/20-displaylink.conf
+%{_presetdir}/95-displaylink.preset
+
+%post config
+%systemd_post displaylink-service
+
+
+
+%prep
+%setup -q -T -D -a 1 -c -n %{name}-rpm-%{version}
+for i in displaylink.service	\
+	displaylink.logrotate	\
+	95-displaylink.preset	\
+	99-displaylink.rules	\
+	20-displaylink.conf	\
+	displaylink-sleep-extractor.sh
+do
+  cp -v displaylink-rpm-%{version}/$i .
+done
+unzip "%{SOURCE0}"
+chmod +x displaylink-driver-%{version}-55.153.run
+./displaylink-driver-%{version}-55.153.run --noexec --keep
+chmod 644 displaylink-driver-%{version}-55.153/LICENSE
+
+%install
+mkdir -p %{buildroot}%{_libexecdir}/%{name}/			\
+	%{buildroot}%{_unitdir}/				\
+	%{buildroot}%{_presetdir}/				\
+	%{buildroot}%{_prefix}/lib/systemd/system-sleep/	\
+	%{buildroot}%{_sysconfdir}/logrotate.d/			\
+	%{buildroot}%{_udevrulesdir}/				\
+	%{buildroot}%{_datadir}/X11/xorg.conf.d/		\
+	%{buildroot}%{_localstatedir}/log/%{name}/
+
+# DisplayLinkManager
+pushd displaylink-driver-%{version}-55.153
+
+cp LICENSE ..
+
+%ifarch x86_64
+cp -a x64-ubuntu-1604/DisplayLinkManager %{buildroot}%{_libexecdir}/%{name}/
+%endif
+
+%ifarch %ix86
+cp -a x86-ubuntu-1604/DisplayLinkManager %{buildroot}%{_libexecdir}/%{name}/
+%endif
+
+%ifarch armv7hl
+cp -a arm-linux-gnueabihf/DisplayLinkManager %{buildroot}%{_libexecdir}/%{name}/
+%endif
+
+# Firmwares
+cp -a ella-dock-release.spkg firefly-monitor-release.spkg ridge-dock-release.spkg %{buildroot}%{_libexecdir}/%{name}/
+
+# pm-util
+bash ../displaylink-sleep-extractor.sh displaylink-installer.sh > %{buildroot}%{_prefix}/lib/systemd/system-sleep/displaylink.sh
+
+chmod +x %{buildroot}%{_prefix}/lib/systemd/system-sleep/displaylink.sh
+
+popd
+
+# systemd/udev
+cp -a displaylink.service %{buildroot}%{_unitdir}
+cp -a displaylink.logrotate %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
+cp -a 95-displaylink.preset %{buildroot}%{_presetdir}
+cp -a 99-displaylink.rules %{buildroot}%{_udevrulesdir}
+cp -a 20-displaylink.conf %{buildroot}%{_datadir}/X11/xorg.conf.d
+
+%changelog
```

Comment:
Shouldn't we keep the changelog from former versions, and maybe add an `Obsoletes:` for the old package?

Reply:
I'm not sure on that. It's largely based on the DKMS displaylink.spec (so yes), but it's also a separate file (so no). Overall, I think it depends on whether this replaces the DKMS displaylink package or not. I suppose you could 'upgrade' or 'downgrade' between the DKMS and akmod, but I haven't and cannot test that (I'm running Silverblue, so DKMS isn't possible, hence this PR).